compiler/rustc_middle/src/ty/generics.rs

```diff
@@ -228,8 +228,43 @@ impl<'tcx> Generics {
         })
     }
 
+    /// Returns the substs corresponding to the generic parameters
+    /// of this item, excluding `Self`.
+    pub fn own_substs_no_defaults(
```

Comment:
Let's make it clear in the naming and docs that this is only meant for use in diagnostics code.

Reply:
added a comment, don't want to make these method names even more unwieldy :sweat_smile: